### PR TITLE
Wojownicy kodu us3

### PR DESCRIPTION
**Pull request by adamvm**

### CODE REVIEW

File: src/CourseDashboard.cpp [hunk 1]
```diff
@@ -82,3 +83,40 @@ void CourseDashboard::updateUser(User & user)
...
+{
+    std::string login, password, loginKeyword, passwordKeyword;
+
+    std::cout << "UserName: "; std::cin >> loginKeyword;
+    std::cout << "Password: "; std::cin >> passwordKeyword;
+
+    std::fstream file;
+    file.open("plik.txt", std::ios::in);
+    //std::getline(file,login);
+    //std::getline(file,password);
```
Comment (adamvm) [inline src/CourseDashboard.cpp]: No comments should be in production code

File: src/CourseDashboard.cpp [hunk 2]
```diff
@@ -82,3 +83,40 @@ void CourseDashboard::updateUser(User & user)
...
+    {
+        std::cout<<"file doesn't exist";
+        exit(0);
+    }
+
+    while(!file.eof())
+    {
+    file>>login;
+    file>>password;
+    //file.open("plik.txt", std::ios::in);
```
Comment (adamvm) [inline src/CourseDashboard.cpp]: No comments should be in production code

File: src/CourseDashboard.cpp [hunk 3]
```diff
@@ -82,3 +83,40 @@ void CourseDashboard::updateUser(User & user)
...
+    std::cout << "UserName: "; std::cin >> loginKeyword;
+    std::cout << "Password: "; std::cin >> passwordKeyword;
+
+    std::fstream file;
+    file.open("plik.txt", std::ios::in);
+    //std::getline(file,login);
+    //std::getline(file,password);
+    if(!file.good())
+    {
+        std::cout<<"file doesn't exist";
```
Comment (adamvm) [inline src/CourseDashboard.cpp]: !file.good can probably mean more different problems than only not existing file

File: src/main.cpp [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
+#include <fstream>
 #include "CourseDashboard.hpp"
 
 
 int main()
 {
     CourseDashboard cdb;
     std::cout << "Course Dashboard" << std::endl;
+    cdb.logIn();
     User u1("Szymon", "SzymonGajewski", "wieczorowa", "SzymonGajewski", "SzymonGajewski");
```
Comment (adamvm) [inline src/main.cpp]: It's better to use fictional data for testing project

**Review by adamvm — COMMENTED** [review]
Please introduce requested changes

File: src/CourseDashboard.cpp (hunk 1 above)
Comment (TheDude-cpu) [reply src/CourseDashboard.cpp]: already in second commit this comments don't exist

File: src/CourseDashboard.cpp [hunk 5]
```diff
@@ -82,3 +83,67 @@ void CourseDashboard::updateUser(User & user)
         break;
     }
 }
+
+std::string CourseDashboard::eMail()
```
Comment (ziobron) [inline src/CourseDashboard.cpp]: it should rather be called `provideEmail`

File: src/CourseDashboard.cpp [hunk 6]
```diff
@@ -82,3 +83,67 @@ void CourseDashboard::updateUser(User & user)
...
+{
+        std::string email;
+        std::cout<<"Please enter a email."<<std::endl;
+        std::cin>>email;
+        std::cout << "e-mail: \"" << email<< "\"\n";
+
+    return email;
+}
+
+std::string CourseDashboard::passWord()
```
Comment (ziobron) [inline src/CourseDashboard.cpp]: providePassword

File: src/CourseDashboard.cpp [hunk 7]
```diff
@@ -82,3 +83,67 @@ void CourseDashboard::updateUser(User & user)
...
+}
+
+void CourseDashboard::logIn()
+{
+    std::string emailKeyword = eMail();
+    std::string passwordKeyword = passWord();
+    std::string email, password;
+    std::string line;
+    std::fstream file;
+    file.open("plik.txt", std::ios::in);
```
Comment (ziobron) [inline src/CourseDashboard.cpp]: line 123 and 124 can be joined. Please use ifstream for reading only.

File: src/CourseDashboard.cpp [hunk 8]
```diff
@@ -82,3 +83,67 @@ void CourseDashboard::updateUser(User & user)
...
+
+    return password1; 
+}
+
+void CourseDashboard::logIn()
+{
+    std::string emailKeyword = eMail();
+    std::string passwordKeyword = passWord();
+    std::string email, password;
+    std::string line;
```
Comment (ziobron) [inline src/CourseDashboard.cpp]: `line` is not used?

File: src/CourseDashboard.cpp (hunk 3 above)
Comment (ziobron) [reply src/CourseDashboard.cpp]: And how to test it? Exception would be a better choice.

File: src/CourseDashboard.cpp [hunk 9]
```diff
@@ -82,3 +83,67 @@ void CourseDashboard::updateUser(User & user)
...
+	}
+
+    return password1; 
+}
+
+void CourseDashboard::logIn()
+{
+    std::string emailKeyword = eMail();
+    std::string passwordKeyword = passWord();
+    std::string email, password;
```
Comment (ziobron) [inline src/CourseDashboard.cpp]: those strings are used inside while loop, so they should be declared there.

File: src/CourseDashboard.cpp [hunk 10]
```diff
@@ -82,3 +83,67 @@ void CourseDashboard::updateUser(User & user)
...
+    }
+
+    while(!file.eof())
+    {
+        file>>email;
+        file>>password;
+
+        if(emailKeyword == email && passwordKeyword == password) 
+        {   
+            std::cout << "Login succeesful!" << std::endl;
```
Comment (ziobron) [inline src/CourseDashboard.cpp]: how can I test if a login was successful or not? proper return value would be more useful.

**Review by ziobron — CHANGES_REQUESTED** [review]
No tests! 👎

File: src/CourseDashboard.cpp (hunk 5 above)
Comment (adamvm) [reply src/CourseDashboard.cpp]: General rule: method should have some verb (czasownik) in the name shortly describing whats method do

File: src/CourseDashboard.cpp [hunk 11]
```diff
@@ -2,83 +2,148 @@
...
+
+	std::ifstream file("plik.txt", std::ios::in);
+
+	if (!file.good())
+	{
+		std::cout << "file doesn't exist";
+		exit(0);
+	}
+
+	if ((provideEmail(emailKeyword) == false) || (providePassword(passwordKeyword) == false))
```
Comment (ziobron) [inline src/CourseDashboard.cpp:132]: || will not work properly, && should be used.

File: src/main.cpp [hunk 12]
```diff
@@ -1,16 +1,18 @@
...
+#include <fstream>
 #include "CourseDashboard.hpp"
 
 
 int main()
 {
     CourseDashboard cdb;
     std::cout << "Course Dashboard" << std::endl;
-    User u1("Szymon", "SzymonGajewski", "wieczorowa", "SzymonGajewski", "SzymonGajewski");
+    cdb.logIn();
```
Comment (ziobron) [inline src/main.cpp:10]: You need to check if the login was successful

File: tests/CourseDashboardTests.cpp [hunk 13]
```diff
@@ -1,8 +1,28 @@
...
+
+    ASSERT_TRUE(result);
+}
+
+TEST_F(TestLogIn, PasswordReturnTrue)
+{
+    auto result = login.providePassword("testForEmail");
+
+    ASSERT_TRUE(result);
+}
```
Comment (ziobron) [inline tests/CourseDashboardTests.cpp:27]: tests for `login()` function are missing

**Review by ziobron — COMMENTED** [review]
There are some tests, but still some of them are missing.

**Comment by ziobron** [comment]
+8 points (some more tests would be useful)

**Pull request closed without merging**